rules/REQUEST-901-INITIALIZATION.conf

```diff
@@ -425,3 +425,35 @@ SecRule TX:detection_paranoia_level "@lt %{tx.blocking_paranoia_level}" \
     log,\
     msg:'Detection paranoia level configured is lower than the paranoia level itself. This is illegal. Blocking request. Aborting',\
     ver:'OWASP_CRS/4.0.0-rc2'"
+
+
+#
+# Engine Variables by CRS
+#
+
+# tx.request_filename
+#
+# Variable REQUEST_FILENAME on libmodsecurity behaves differently than ModSecurity v2
+# tx.request_filename fixes this misbehavior by creating a new version of REQUEST_FILENAME
+#
+# Example:
+#   http://example.com/index.php?param=2
+#   tx.request_filename = /index.php
+#
+SecRule REQUEST_URI_RAW "@rx ^([^?]+)" \
+    "id:901600,\
+    phase:1,\
+    pass,\
+    capture,\
+    t:none,\
+    nolog,\
+    setvar:'tx.tmp_request_filename=%{tx.1}'"
+
+SecRule REQUEST_FILENAME "!@streq %{tx.tmp_request_filename}" \
+    "id:901610,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    setvar:'tx.request_filename=%{tx.tmp_request_filename}',\
+    setvar:'tx.tmp_request_filename=0'"
```

Comment:
Are you using `0` here on purpose? Is that better than empty?

Reply:
I think lint doesn't allow me to do it